.github/workflows/python-package.yml

```diff
@@ -31,13 +31,13 @@ jobs:
         python -m pip install .[arrays,test]
     - name: Test source code and docs
       run: |
-        cd tests
-        python -m pytest --cov=uncertainties --cov=. --cov-report=xml --cov-report=term
+        python -m pytest --cov --cov-report xml --cov-report term
     - name: Upload coverage reports to Codecov
-      uses: codecov/codecov-action@v4.0.1
+      uses: codecov/codecov-action@v4.6.0
       with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        slug: lmfit/uncertainties
+        file: ./coverage.xml
```

Comment:
Could we add a `name` entry here as well? If we set the name based on the Python version and OS, would that let us tell which report was from which run? Right now all the reports for PR have the same name on the Codecov page so it is hard to tell which is which. This mainly matters for distinguishing the no-numpy run from everything else.

Reply:
I think `tag`s are what we're looking for here. Yes, we can do that. It's been my plan to add tags, but I wanted to get the more basic configuration working. I doubt adding tags will fix the issues we're seeing, but maybe I'm wrong. Maybe the tags will help.